### ISSUE TITLE
gatk error after last night's (successful) upgrade

### ISSUE THREAD

**Issue by naumenko-sa** [chapmanb/bcbio-nextgen#1689]
Hello!

After successful upgrade last night with
`bcbio_nextgen.py upgrade -u development --tools --data`

I have an error:
`gatk -R ~/work/tools/bcbio/genomes/Hsapiens/GRCh37/seq/GRCh37.fa -T VariantsToTable -V 394-gatk-haplotype.decomposed.vcf.gz -F CHROM -F POS -F REF -F ALT -F DP -GF DP -GF AD -o test.table --allowMissingData`
```
'import site' failed; use -v for traceback
Traceback (most recent call last):
  File "/home/naumenko/work/tools/bin/gatk", line 9, in <module>
    import os
  File "/home/naumenko/work/tools/bcbio/anaconda/lib/python2.7/os.py", line 400, in <module>
    import UserDict
  File "/home/naumenko/work/tools/bcbio/anaconda/lib/python2.7/UserDict.py", line 116, in <module>
    import _abcoll
  File "/home/naumenko/work/tools/bcbio/anaconda/lib/python2.7/_abcoll.py", line 456, in <module>
    KeysView.register(type({}.viewkeys()))
AttributeError: 'dict' object has no attribute 'viewkeys'
```

Without wrapper it works:
`java -Xmx1g -jar ~/work/tools/bcbio/toolplus/gatk/3.6-0-g89b7209/GenomeAnalysisTK.jar -R ~/work/tools/bcbio/genomes/Hsapiens/GRCh37/seq/GRCh37.fa -T VariantsToTable -V 394-gatk-haplotype.decomposed.vcf.gz -F CHROM -F POS -F REF -F ALT -F DP -GF DP -GF AD -o test.table --allowMissingData`
```
INFO  11:41:15,227 HelpFormatter - --------------------------------------------------------------------------------- 
INFO  11:41:15,231 HelpFormatter - The Genome Analysis Toolkit (GATK) v3.6-0-g89b7209, Compiled 2016/06/01 22:27:29 
INFO  11:41:15,234 HelpFormatter - Copyright (c) 2010-2016 The Broad Institute 
INFO  11:41:15,234 HelpFormatter - For support and documentation go to https://www.broadinstitute.org/gatk 
INFO  11:41:15,234 HelpFormatter - [Tue Dec 13 11:41:15 EST 2016] Executing on Linux 2.6.32-642.6.2.el6.x86_64 amd64 
INFO  11:41:15,235 HelpFormatter - Java HotSpot(TM) 64-Bit Server VM 1.8.0_91-b14 JdkDeflater 
INFO  11:41:15,238 HelpFormatter - Program Args: -R /home/naumenko/work/tools/bcbio/genomes/Hsapiens/GRCh37/seq/GRCh37.fa -T VariantsToTable -V 394-gatk-haplotype.decomposed.vcf.gz -F CHROM -F POS -F REF -F ALT -F DP -GF DP -GF AD -o test.table --allowMissingData 
INFO  11:41:15,264 HelpFormatter - Executing as naumenko@qlogin1 on Linux 2.6.32-642.6.2.el6.x86_64 amd64; Java HotSpot(TM) 64-Bit Server VM 1.8.0_91-b14. 
INFO  11:41:15,265 HelpFormatter - Date/Time: 2016/12/13 11:41:15 
INFO  11:41:15,265 HelpFormatter - --------------------------------------------------------------------------------- 
INFO  11:41:15,266 HelpFormatter - --------------------------------------------------------------------------------- 
INFO  11:41:15,298 GenomeAnalysisEngine - Strictness is SILENT 
INFO  11:41:15,734 GenomeAnalysisEngine - Downsampling Settings: Method: BY_SAMPLE, Target Coverage: 1000 
WARN  11:41:15,977 IndexDictionaryUtils - Track variant doesn't have a sequence dictionary built in, skipping dictionary validation 
INFO  11:41:16,178 GenomeAnalysisEngine - Preparing for traversal 
INFO  11:41:16,192 GenomeAnalysisEngine - Done preparing for traversal 
INFO  11:41:16,194 ProgressMeter - [INITIALIZATION COMPLETE; STARTING PROCESSING] 
INFO  11:41:16,194 ProgressMeter -                 | processed |    time |    per 1M |           |   total | remaining 
INFO  11:41:16,194 ProgressMeter -        Location |     sites | elapsed |     sites | completed | runtime |   runtime 
INFO  11:41:36,677 ProgressMeter -            done    369157.0    20.0 s      55.0 s      100.0%    20.0 s       0.0 s 
INFO  11:41:36,678 ProgressMeter - Total runtime 20.48 secs, 0.34 min, 0.01 hours 
------------------------------------------------------------------------------------------
Done. There were 1 WARN messages, the first 1 are repeated below.
WARN  11:41:15,977 IndexDictionaryUtils - Track variant doesn't have a sequence dictionary built in, skipping dictionary validation 
```

Before the update I used the wrapper for many times. 
Could you please point out what is happening?
May this cause problems with variant calling in the variant2 pipeline?

Thanks!
Sergey

**Comment by chapmanb** [comment]
Sergey;
Sorry about the problem, I believe the issue was that the new wrapper around GATK uses Python and did not explicitly specify the path to the anaconda installed python. It likely picked up a different version and caused incompatibility issues. I pushed a fix that resolves this (and also bumps GATK to the recently released 3.7). You can update the wrapper with:
```
bcbio_conda install -c bioconda gatk
```
and will need to re-run gatk-register (you can use `--noversioncheck` if you want to use something older than 3.7). Hope this fixes things for you.

**Comment by naumenko-sa** [comment]
Thanks Brad, now it works.
